src/app/_components/InputFields/TextInputField.js

```diff
@@ -0,0 +1,98 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import InputErrorIcon from '../Icons/InpurErrorIcon';
+import { cn } from '@/utils/cn';
+import { variants } from './styles';
+
+export function TextInputField({
+  value,
+  name,
+  onChange,
+  disabled = false,
+  placeholder = '',
+  error = '',
+  required = false,
+  variant = variants.default,
+}) {
+  const id = `textinput_${name}`;
+  return (
+    <div className={cn(variant.mainContainer.base)}>
+      {error && <p className={cn(variant.errorParagraph.base)}>{error}</p>}
+      <div
+        className={cn(
+          variant.inputContainer.base,
+          variant.inputContainer.style,
+          variant.inputContainer.focusWithin,
+          error && variant.inputContainer.error,
+        )}
+      >
+        <input
+          className={cn(
+            variant.input.base,
+            variant.input.focus,
+            variant.input.style,
+            error && variant.input.error,
+          )}
+          id={id}
+          type="text"
```

Comment:
@IvanKoshan Consider making this component more flexible to cover also other than `text` types. Eg. `email`, `password`, `url`. You can pass `type` as a property and define needed list of possible values in the `propTypes`. "text" could be a defaultProps value.

Reply:
did it